### PR TITLE
Fix build with Typescript v4

### LINKED ISSUE #234: Error compiling with Typescript v4
```
node_modules/microsoft-cognitiveservices-speech-sdk/distrib/lib/src/sdk/CustomCommandsConfig.d.ts(45,9): error TS2611: 'applicationId' is defined as a property in class 'DialogServiceConfigImpl', but is overridden here in 'CustomCommandsConfig' as an accessor.
```

### PR DESCRIPTION
**Pull request by orgads**
Fixes #234

### CODE REVIEW

**Comment by orgads** [comment]
@rhurey This blocks us from integrating the package. Please merge and release ASAP. Thanks.

File: src/common/Promise.ts [hunk 1]
```diff
@@ -116,8 +116,8 @@ export class Sink<T> {
     private privPromiseResult: PromiseResult<T> = null;
     private privPromiseResultEvents: PromiseResultEventSource<T> = null;
 
-    private privSuccessHandlers: Array<((result: T) => void)> = [];
-    private privErrorHandlers: Array<(e: string) => void> = [];
+    private privSuccessHandlers: ((result: T) => void)[] = [];
```
Comment (rhurey) [inline src/common/Promise.ts:119]: Lint rules are failing with: ERROR: D:/a/1/s/src/common/Promise.ts:119:34 - Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.

Comment (orgads) [reply src/common/Promise.ts:119]: Thanks, fixed.

**Comment by orgads** [comment]
ping

**Comment by orgads** [comment]
@rhurey/@glharper ping

**Comment by rhurey** [comment]
I'm still getting linting failures.
C:\git\cognitive-services-speech-sdk-js>git fetch origin pull/236/head:pull236
...
C:\git\cognitive-services-speech-sdk-js>git checkout pull236
Switched to branch 'pull236'

C:\git\cognitive-services-speech-sdk-js>npm run build

> microsoft-cognitiveservices-speech-sdk@1.14.0-alpha.0.1 build C:\git\cognitive-services-speech-sdk-js
> gulp compress && gulp build2015

[15:51:31] Using gulpfile C:\git\cognitive-services-speech-sdk-js\gulpfile.js
[15:51:31] Starting 'compress'...
[15:51:31] Starting 'bundle'...
[15:51:31] Starting 'build'...
[15:51:31] Starting 'build'...
Warning: The 'no-floating-promises' rule requires type information.

ERROR: C:/git/cognitive-services-speech-sdk-js/src/common/Promise.ts:119:34 - Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.
ERROR: C:/git/cognitive-services-speech-sdk-js/src/common/Promise.ts:120:32 - Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.


ERROR: C:/git/cognitive-services-speech-sdk-js/src/common/Queue.ts:120:37 - Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.

**Comment by orgads** [comment]
Sorry, pushed from an old branch. Fixed now.

**Review by rhurey — APPROVED** [review]

**Pull request merged**